include/dqlite.h

```diff
@@ -17,8 +17,304 @@
 
 int dqlite_version_number(void);
 
+/**
+ * Hold the value of a dqlite node ID. Guaranteed to be at least 64-bit long.
+ */
+typedef unsigned long long dqlite_node_id;
+
+typedef struct dqlite_server dqlite_server;
+typedef struct dqlite dqlite;
+typedef struct dqlite_stmt dqlite_stmt;
+
+/**
+ * Signature of a custom callback used to establish network connections
+ * to dqlite servers.
+ *
+ * @arg is a user data parameter, copied from the third argument of
+ * dqlite_server_set_connect_func. @addr is a (borrowed) abstract address
+ * string, as passed to dqlite_server_create or dqlite_server_set_peer. @fd is
+ * an address where a socket representing the connection should be stored. The
+ * callback should return zero if a connection was established successfully or
+ * nonzero if the attempt failed.
+ */
+typedef int (*dqlite_connect_func)(void *arg, const char *addr, int *fd);
+
+/* The following functions (dqlite_server_* and the functions that mimic the
+ * sqlite3 API) use SQLite result codes. dqlite tries to return result codes
+ * that match what SQLite would return in the same situation, but in some cases
```

Comment:
Returning SQLite result codes surely makes sense for the functions that mimic the sqlite3 API.

However I'm not sure about the `dqlite_server` ones. It feels we might end up shoehorning them, since the failure modes of the `dqlite_server` APIs are a bit different and if we need to **add** ad-hoc codes for `dqlite_server`, then having a different set altogether might be a saner choice.

This is a detail that can be revisited later (either in one way or the other) when things become more clear.